node/pkg/processor/processor.go

```diff
@@ -429,3 +444,35 @@ func (p *Processor) vaaWriter(ctx context.Context) error {
 		}
 	}
 }
+
+func (p *Processor) processTssSignature(sig *tsscommon.SignatureData) {
+	if sig == nil {
+		return
+	}
+
+	if sig.Signature == nil {
+		p.logger.Error("received TSS signature with nil signature")
+		return
+	}
+
+	if sig.M == nil {
+		p.logger.Error("received TSS signature with nil message")
+		return
+	}
+
+	digestBytes := sig.M
+	hash := hex.EncodeToString(digestBytes)
+	wtr, ok := p.tssWaiters[hash]
+	if !ok {
+		// TODO: this indicates a TSS signature that was waited for too long probably.
+		p.logger.Warn("received TSS signature for unknown VAA", zap.String("hash", hash))
+		return
+	}
+
+	// TODO: Should we return a signature already in the correct format from the TssEngine? or giving the processor more control is better?
```

Comment:
That is, the signature we output cannot be processed on a smart contract without appending to it the recovery bit at the end (which i do  a line or two after the comment above).

Instead, currently i output tss.Signature which contains everything needed to modify the bytes such that it can be verified on a smart contract (requires more knowledge from the user).

Reply:
I see, I think it's okay as it is then